File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-eq-shared-5-4/Representations.cpp
// Representations.cpp 
#include "Representations.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

Representations::Representations(TiXmlNode* xmlNode)
: ISerialized(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //nonRelianceNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* nonRelianceNode = xmlNode->FirstChildElement("nonReliance");

   if(nonRelianceNode){nonRelianceIsNull_ = false;}
   else{nonRelianceIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- nonRelianceNode , address : " << nonRelianceNode << std::endl;
   #endif
   if(nonRelianceNode)
   {
      if(nonRelianceNode->Attribute("href") || nonRelianceNode->Attribute("id"))
      {
          if(nonRelianceNode->Attribute("id"))
          {
             nonRelianceIDRef_ = nonRelianceNode->Attribute("id");
             nonReliance_ = boost::shared_ptr<XsdTypeBoolean>(new XsdTypeBoolean(nonRelianceNode));
             nonReliance_->setID(nonRelianceIDRef_);
             IDManager::instance().setID(nonRelianceIDRef_,nonReliance_);
          }
          else if(nonRelianceNode->Attribute("href")) { nonRelianceIDRef_ = nonRelianceNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { nonReliance_ = boost::shared_ptr<XsdTypeBoolean>(new XsdTypeBoolean(nonRelianceNode));}
   }

   //agreementsRegardingHedgingNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* agreementsRegardingHedgingNode = xmlNode->FirstChildElement("agreementsRegardingHedging");

   if(agreementsRegardingHedgingNode){agreementsRegardingHedgingIsNull_ = false;}
   else{agreementsRegardingHedgingIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- agreementsRegardingHedgingNode , address : " << agreementsRegardingHedgingNode << std::endl;
   #endif
   if(agreementsRegardingHedgingNode)
   {
      if(agreementsRegardingHedgingNode->Attribute("href") || agreementsRegardingHedgingNode->Attribute("id"))
      {
          if(agreementsRegardingHedgingNode->Attribute("id"))
          {
             agreementsRegardingHedgingIDRef_ = agreementsRegardingHedgingNode->Attribute("id");
             agreementsRegardingHedging_ = boost::shared_ptr<XsdTypeBoolean>(new XsdTypeBoolean(agreementsRegardingHedgingNode));
             agreementsRegardingHedging_->setID(agreementsRegardingHedgingIDRef_);
             IDManager::instance().setID(agreementsRegardingHedgingIDRef_,agreementsRegardingHedging_);
          }
          else if(agreementsRegardingHedgingNode->Attribute("href")) { agreementsRegardingHedgingIDRef_ = agreementsRegardingHedgingNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { agreementsRegardingHedging_ = boost::shared_ptr<XsdTypeBoolean>(new XsdTypeBoolean(agreementsRegardingHedgingNode));}
   }

   //indexDisclaimerNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* indexDisclaimerNode = xmlNode->FirstChildElement("indexDisclaimer");

   if(indexDisclaimerNode){indexDisclaimerIsNull_ = false;}
   else{indexDisclaimerIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- indexDisclaimerNode , address : " << indexDisclaimerNode << std::endl;
   #endif
   if(indexDisclaimerNode)
   {
      if(indexDisclaimerNode->Attribute("href") || indexDisclaimerNode->Attribute("id"))
      {
          if(indexDisclaimerNode->Attribute("id"))
          {
             indexDisclaimerIDRef_ = indexDisclaimerNode->Attribute("id");
             indexDisclaimer_ = boost::shared_ptr<XsdTypeBoolean>(new XsdTypeBoolean(indexDisclaimerNode));
             indexDisclaimer_->setID(indexDisclaimerIDRef_);
             IDManager::instance().setID(indexDisclaimerIDRef_,indexDisclaimer_);
          }
          else if(indexDisclaimerNode->Attribute("href")) { indexDisclaimerIDRef_ = indexDisclaimerNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { indexDisclaimer_ = boost::shared_ptr<XsdTypeBoolean>(new XsdTypeBoolean(indexDisclaimerNode));}
   }

   //additionalAcknowledgementsNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* additionalAcknowledgementsNode = xmlNode->FirstChildElement("additionalAcknowledgements");

   if(additionalAcknowledgementsNode){additionalAcknowledgementsIsNull_ = false;}
   else{additionalAcknowledgementsIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- additionalAcknowledgementsNode , address : " << additionalAcknowledgementsNode << std::endl;
   #endif
   if(additionalAcknowledgementsNode)
   {
      if(additionalAcknowledgementsNode->Attribute("href") || additionalAcknowledgementsNode->Attribute("id"))
      {
          if(additionalAcknowledgementsNode->Attribute("id"))
          {
             additionalAcknowledgementsIDRef_ = additionalAcknowledgementsNode->Attribute("id");
             additionalAcknowledgements_ = boost::shared_ptr<XsdTypeBoolean>(new XsdTypeBoolean(additionalAcknowledgementsNode));
             additionalAcknowledgements_->setID(additionalAcknowledgementsIDRef_);
             IDManager::instance().setID(additionalAcknowledgementsIDRef_,additionalAcknowledgements_);
          }
          else if(additionalAcknowledgementsNode->Attribute("href")) { additionalAcknowledgementsIDRef_ = additionalAcknowledgementsNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { additionalAcknowledgements_ = boost::shared_ptr<XsdTypeBoolean>(new XsdTypeBoolean(additionalAcknowledgementsNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<XsdTypeBoolean> Representations::getNonReliance()
{
   if(!this->nonRelianceIsNull_){
        if(nonRelianceIDRef_ != ""){
             return boost::shared_static_cast<XsdTypeBoolean>(IDManager::instance().getID(nonRelianceIDRef_));
        }else{
             return this->nonReliance_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeBoolean>();
   }
}
boost::shared_ptr<XsdTypeBoolean> Representations::getAgreementsRegardingHedging()
{
   if(!this->agreementsRegardingHedgingIsNull_){
        if(agreementsRegardingHedgingIDRef_ != ""){
             return boost::shared_static_cast<XsdTypeBoolean>(IDManager::instance().getID(agreementsRegardingHedgingIDRef_));
        }else{
             return this->agreementsRegardingHedging_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeBoolean>();
   }
}
boost::shared_ptr<XsdTypeBoolean> Representations::getIndexDisclaimer()
{
   if(!this->indexDisclaimerIsNull_){
        if(indexDisclaimerIDRef_ != ""){
             return boost::shared_static_cast<XsdTypeBoolean>(IDManager::instance().getID(indexDisclaimerIDRef_));
        }else{
             return this->indexDisclaimer_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeBoolean>();
   }
}
boost::shared_ptr<XsdTypeBoolean> Representations::getAdditionalAcknowledgements()
{
   if(!this->additionalAcknowledgementsIsNull_){
        if(additionalAcknowledgementsIDRef_ != ""){
             return boost::shared_static_cast<XsdTypeBoolean>(IDManager::instance().getID(additionalAcknowledgementsIDRef_));
        }else{
             return this->additionalAcknowledgements_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeBoolean>();
   }
}
}

File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-eq-shared-5-4/Representations.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class Representations : ISerialized
    {
        public Representations(XmlNode xmlNode)
        {
            XmlNode nonRelianceNode = xmlNode.SelectSingleNode("nonReliance");
            
            if (nonRelianceNode != null)
            {
                if (nonRelianceNode.Attributes["href"] != null || nonRelianceNode.Attributes["id"] != null) 
                {
                    if (nonRelianceNode.Attributes["id"] != null) 
                    {
                        nonRelianceIDRef_ = nonRelianceNode.Attributes["id"].Value;
                        XsdTypeBoolean ob = new XsdTypeBoolean(nonRelianceNode);
                        IDManager.SetID(nonRelianceIDRef_, ob);
                    }
                    else if (nonRelianceNode.Attributes["href"] != null)
                    {
                        nonRelianceIDRef_ = nonRelianceNode.Attributes["href"].Value;
                    }
                    else
                    {
                        nonReliance_ = new XsdTypeBoolean(nonRelianceNode);
                    }
                }
                else
                {
                    nonReliance_ = new XsdTypeBoolean(nonRelianceNode);
                }
            }
            
        
            XmlNode agreementsRegardingHedgingNode = xmlNode.SelectSingleNode("agreementsRegardingHedging");
            
            if (agreementsRegardingHedgingNode != null)
            {
                if (agreementsRegardingHedgingNode.Attributes["href"] != null || agreementsRegardingHedgingNode.Attributes["id"] != null) 
                {
                    if (agreementsRegardingHedgingNode.Attributes["id"] != null) 
                    {
                        agreementsRegardingHedgingIDRef_ = agreementsRegardingHedgingNode.Attributes["id"].Value;
                        XsdTypeBoolean ob = new XsdTypeBoolean(agreementsRegardingHedgingNode);
                        IDManager.SetID(agreementsRegardingHedgingIDRef_, ob);
                    }
                    else if (agreementsRegardingHedgingNode.Attributes["href"] != null)
                    {
                        agreementsRegardingHedgingIDRef_ = agreementsRegardingHedgingNode.Attributes["href"].Value;
                    }
                    else
                    {
                        agreementsRegardingHedging_ = new XsdTypeBoolean(agreementsRegardingHedgingNode);
                    }
                }
                else
                {
                    agreementsRegardingHedging_ = new XsdTypeBoolean(agreementsRegardingHedgingNode);
                }
            }
            
        
            XmlNode indexDisclaimerNode = xmlNode.SelectSingleNode("indexDisclaimer");
            
            if (indexDisclaimerNode != null)
            {
                if (indexDisclaimerNode.Attributes["href"] != null || indexDisclaimerNode.Attributes["id"] != null) 
                {
                    if (indexDisclaimerNode.Attributes["id"] != null) 
                    {
                        indexDisclaimerIDRef_ = indexDisclaimerNode.Attributes["id"].Value;
                        XsdTypeBoolean ob = new XsdTypeBoolean(indexDisclaimerNode);
                        IDManager.SetID(indexDisclaimerIDRef_, ob);
                    }
                    else if (indexDisclaimerNode.Attributes["href"] != null)
                    {
                        indexDisclaimerIDRef_ = indexDisclaimerNode.Attributes["href"].Value;
                    }
                    else
                    {
                        indexDisclaimer_ = new XsdTypeBoolean(indexDisclaimerNode);
                    }
                }
                else
                {
                    indexDisclaimer_ = new XsdTypeBoolean(indexDisclaimerNode);
                }
            }
            
        
            XmlNode additionalAcknowledgementsNode = xmlNode.SelectSingleNode("additionalAcknowledgements");
            
            if (additionalAcknowledgementsNode != null)
            {
                if (additionalAcknowledgementsNode.Attributes["href"] != null || additionalAcknowledgementsNode.Attributes["id"] != null) 
                {
                    if (additionalAcknowledgementsNode.Attributes["id"] != null) 
                    {
                        additionalAcknowledgementsIDRef_ = additionalAcknowledgementsNode.Attributes["id"].Value;
                        XsdTypeBoolean ob = new XsdTypeBoolean(additionalAcknowledgementsNode);
                        IDManager.SetID(additionalAcknowledgementsIDRef_, ob);
                    }
                    else if (additionalAcknowledgementsNode.Attributes["href"] != null)
                    {
                        additionalAcknowledgementsIDRef_ = additionalAcknowledgementsNode.Attributes["href"].Value;
                    }
                    else
                    {
                        additionalAcknowledgements_ = new XsdTypeBoolean(additionalAcknowledgementsNode);
                    }
                }
                else
                {
                    additionalAcknowledgements_ = new XsdTypeBoolean(additionalAcknowledgementsNode);
                }
            }
            
        
        }
        
    
        #region NonReliance_
        private XsdTypeBoolean nonReliance_;
        public XsdTypeBoolean NonReliance_
        {
            get
            {
                if (this.nonReliance_ != null)
                {
                    return this.nonReliance_; 
                }
                else if (this.nonRelianceIDRef_ != null)
                {
                    nonReliance_ = IDManager.getID(nonRelianceIDRef_) as XsdTypeBoolean;
                    return this.nonReliance_; 
                }
                else
                {
                      return this.nonReliance_; 
                }
            }
            set
            {
                if (this.nonReliance_ != value)
                {
                    this.nonReliance_ = value;
                }
            }
        }
        #endregion
        
        public string nonRelianceIDRef_ { get; set; }
        #region AgreementsRegardingHedging_
        private XsdTypeBoolean agreementsRegardingHedging_;
        public XsdTypeBoolean AgreementsRegardingHedging_
        {
            get
            {
                if (this.agreementsRegardingHedging_ != null)
                {
                    return this.agreementsRegardingHedging_; 
                }
                else if (this.agreementsRegardingHedgingIDRef_ != null)
                {
                    agreementsRegardingHedging_ = IDManager.getID(agreementsRegardingHedgingIDRef_) as XsdTypeBoolean;
                    return this.agreementsRegardingHedging_; 
                }
                else
                {
                      return this.agreementsRegardingHedging_; 
                }
            }
            set
            {
                if (this.agreementsRegardingHedging_ != value)
                {
                    this.agreementsRegardingHedging_ = value;
                }
            }
        }
        #endregion
        
        public string agreementsRegardingHedgingIDRef_ { get; set; }
        #region IndexDisclaimer_
        private XsdTypeBoolean indexDisclaimer_;
        public XsdTypeBoolean IndexDisclaimer_
        {
            get
            {
                if (this.indexDisclaimer_ != null)
                {
                    return this.indexDisclaimer_; 
                }
                else if (this.indexDisclaimerIDRef_ != null)
                {
                    indexDisclaimer_ = IDManager.getID(indexDisclaimerIDRef_) as XsdTypeBoolean;
                    return this.indexDisclaimer_; 
                }
                else
                {
                      return this.indexDisclaimer_; 
                }
            }
            set
            {
                if (this.indexDisclaimer_ != value)
                {
                    this.indexDisclaimer_ = value;
                }
            }
        }
        #endregion
        
        public string indexDisclaimerIDRef_ { get; set; }
        #region AdditionalAcknowledgements_
        private XsdTypeBoolean additionalAcknowledgements_;
        public XsdTypeBoolean AdditionalAcknowledgements_
        {
            get
            {
                if (this.additionalAcknowledgements_ != null)
                {
                    return this.additionalAcknowledgements_; 
                }
                else if (this.additionalAcknowledgementsIDRef_ != null)
                {
                    additionalAcknowledgements_ = IDManager.getID(additionalAcknowledgementsIDRef_) as XsdTypeBoolean;
                    return this.additionalAcknowledgements_; 
                }
                else
                {
                      return this.additionalAcknowledgements_; 
                }
            }
            set
            {
                if (this.additionalAcknowledgements_ != value)
                {
                    this.additionalAcknowledgements_ = value;
                }
            }
        }
        #endregion
        
        public string additionalAcknowledgementsIDRef_ { get; set; }
        
    
        
    
    }
    
}

